apollo-client/src/test/java/com/ctrip/framework/apollo/spring/JavaConfigAnnotationTest.java

```diff
@@ -91,6 +108,74 @@ public void testApolloConfigWithInheritance() throws Exception {
     assertEquals(applicationConfig, bean.getSomeConfig());
   }
 
+  @Test
+  public void testEnableApolloConfigResolveExpressionSimple() {
+    String someKey = "someKey-2020-11-14-1750";
+    String someValue = UUID.randomUUID().toString();
+    mockConfig(ConfigConsts.NAMESPACE_APPLICATION, mock(Config.class));
+    Config xxxConfig = mock(Config.class);
+    when(xxxConfig.getProperty(eq(someKey), anyString())).thenReturn(someValue);
+    mockConfig("xxx", xxxConfig);
+
+    TestEnableApolloConfigResolveExpressionWithDefaultValueConfiguration configuration =
+        getSimpleBean(TestEnableApolloConfigResolveExpressionWithDefaultValueConfiguration.class);
+
+    // check
+    assertEquals(someValue, configuration.getSomeKey());
+    verify(xxxConfig, times(1)).getProperty(eq(someKey), anyString());
+  }
+
+  @Test
+  public void testEnableApolloConfigResolveExpressionFromSystemProperty() {
+    mockConfig(ConfigConsts.NAMESPACE_APPLICATION, mock(Config.class));
+    final String someKey = "someKey-2020-11-14-1750";
+    final String someValue = UUID.randomUUID().toString();
+
+    final String propertyKey = "simple.namespace";
+    final String resolvedNamespaceName = "yyy";
+    System.setProperty(propertyKey, resolvedNamespaceName);
+
+    Config yyyConfig = mock(Config.class);
+    when(yyyConfig.getProperty(eq(someKey), anyString())).thenReturn(someValue);
+    mockConfig(resolvedNamespaceName, yyyConfig);
+
+    TestEnableApolloConfigResolveExpressionWithDefaultValueConfiguration configuration =
+        getSimpleBean(TestEnableApolloConfigResolveExpressionWithDefaultValueConfiguration.class);
+
+    // check
+    assertEquals(someValue, configuration.getSomeKey());
+    verify(yyyConfig, times(1)).getProperty(eq(someKey), anyString());
+
+    System.clearProperty(propertyKey);
```

Comment:
Please clear the state in tearDown method

Reply:
Use 
```java
  @After
  public void javaConfigAnnotationTestTearDown() {
    // clear the system property
    clearSystemPropertiesDefineWithStaticStringField(SystemPropertyKeyConstants.class);
  }
```
to do it.

Reply:
This is a little complex, I think we could simply assign some variables and just use System.clearProperty(xxx) in tear down method.

Reply:
Change to code like follow:
```java
  /**
   * forbidden to override the method {@link super#tearDown()}.
   */
  @After
  public void javaConfigAnnotationTestTearDown() {
    // clear the system properties
    System.clearProperty(SystemPropertyKeyConstants.SIMPLE_NAMESPACE);
    System.clearProperty(SystemPropertyKeyConstants.REDIS_NAMESPACE);
    System.clearProperty(SystemPropertyKeyConstants.FROM_SYSTEM_NAMESPACE);
    System.clearProperty(SystemPropertyKeyConstants.FROM_SYSTEM_YAML_NAMESPACE);
    System.clearProperty(SystemPropertyKeyConstants.FROM_NAMESPACE_APPLICATION_KEY);
    System.clearProperty(SystemPropertyKeyConstants.FROM_NAMESPACE_APPLICATION_KEY_YAML);
  }
```
Clear system properties manually instead of using reflection.